test_tools/stdin2rllib/main.cc

```diff
@@ -185,26 +190,29 @@ void parse_and_send_ca_event(json::value obj, r::live_model& rl, r::api_status&
 	}
 }
 
-void parse_and_send_cb_event(json::value obj, r::live_model& rl, r::api_status& status, bool debug) {
+void parse_and_send_cb_event(const rj::Document& obj, r::live_model& rl, r::api_status& status, bool debug) {
 	//parse event id
-	auto evt_id = U("EventId");
-	bool has_event_id = obj.has_field(evt_id);
+	auto evt_id = "EventId";
+	bool has_event_id = obj.HasMember(evt_id);
 	if (!has_event_id) {
 		std::cout << "missing 'EventId' field" << std::endl;
 		return;
 	}
-	std::string event_id = to_utf8string(obj.at(evt_id).as_string());
+	std::string event_id = obj[evt_id].GetString();
 
 	//parse the joined event
-	auto context = U("c");
-	bool has_context = obj.has_field(context);
+	auto context = "c";
+	bool has_context = obj.HasMember(context);
 	if (!has_context) {
 		std::cout << "missing 'context' field" << std::endl;
 		return;
 	}
 	//extract context string
-	json::value v = obj.at(context);
-	std::string c = to_utf8string(v.serialize());
+	auto& context_obj = obj[context];
+	rj::StringBuffer sb;
+	rj::Writer<rj::StringBuffer> writer(sb);
+	context_obj.Accept(writer);
+	std::string c = sb.GetString();
```

Comment:
Similarly, sb.GetString() returns a pointer, need sb.GetSize()? Perhaps just pass the pointer to avoid copying.

Reply:
I think I fixed this, could you take a look? Thanks!

Reply:
LGTM, thanks